sim/paladin/avenging_wrath.go

```diff
@@ -0,0 +1,53 @@
+package paladin
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+)
+
+func (paladin *Paladin) registerAvengingWrath() {
+	actionID := core.ActionID{SpellID: 407788}
+
+	AvengingWrathAura := paladin.RegisterAura(core.Aura{
+		Label:    "Avenging Wrath",
+		ActionID: actionID,
+		Duration: time.Second * 20,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Unit.PseudoStats.DamageDealtMultiplier *= 1.2
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			aura.Unit.PseudoStats.DamageDealtMultiplier /= 1.2
+		},
+	})
+	core.RegisterPercentDamageModifierEffect(AvengingWrathAura, 1.2)
+
+	AvengingWrath := paladin.RegisterSpell(core.SpellConfig{
+		ActionID: actionID,
+		Flags:    core.SpellFlagNoOnCastComplete | core.SpellFlagAPL,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost: 0.08,
+		},
+		Cast: core.CastConfig{
+			CD: core.Cooldown{
+				Timer:    paladin.NewTimer(),
+				Duration: time.Minute*3 - (time.Second * 20),
+			},
+			
+		},
+		ApplyEffects: func(sim *core.Simulation, _ *core.Unit, _ *core.Spell) {
+			AvengingWrathAura.Activate(sim)
+		},
+	})
+
+	paladin.AddMajorCooldown(core.MajorCooldown{
+		Spell: AvengingWrath,
+		Type:  core.CooldownTypeDPS,
+		// modify this logic if it should ever not be spammed on CD / maybe should synced with other CDs
+		ShouldActivate: func(sim *core.Simulation, character *core.Character) bool {
+			
+			return true
```

Comment:
🔨 The sim will auto-execute DPS cooldowns without this when using the `Autocast other cooldowns` action so this isn't necessary. That kind of logic can be handled in an APL if it should be used at a specific time